src/components/ui4/BalanceCardUi4.tsx

```diff
@@ -64,26 +68,31 @@ export const BalanceCardUi4 = (props: Props) => {
   const balanceString = fiatSymbol.length !== 1 ? `${formattedFiat} ${fiatCurrencyCode}` : `${fiatSymbol} ${formattedFiat} ${fiatCurrencyCode}`
   const animateNumber = fiatAmount < MAX_ANIMATED_AMOUNT
 
+  useAsyncEffect(
+    async () => {
+      await account.waitForAllWallets()
+      setIsWalletsLoaded(true)
+    },
+    [],
+    'BalanceCard'
+  )
```

Comment:
I don't think we need `useAsyncEffect` here. Instead, we can consolidate the `exchangeRatesReady` logic with the `isWalletsLoaded` logic into a single loop:

```ts
const activeWalletIds = useWatch(account, 'activeWalletIds')
const currencyWallets = useWatch(account, 'currencyWallets')
const currencyWalletErrors = useWatch(account, 'currencyWalletErrors')

const exchangeRatesReady = React.useMemo(() =>
  activeWalletIds.every(walletId => {
    // Ignore wallets that have crashed:
    if (currencyWalletErrors[walletId]) return true

    // Both the wallet and its rate need to be loaded:
    const wallet = currencyWallets[walletId]
    return wallet != null && exchangeRates[`${wallet.currencyInfo.currencyCode}_${defaultIsoFiat}`] != null
  })
, [activeWalletIds, currencyWallets, currencyWalletErrors])
```